core/src/main/java/org/springframework/security/authorization/method/AuthorizationAdvisorProxyFactory.java

```diff
@@ -165,7 +169,13 @@ public static AuthorizationAdvisorProxyFactory withReactiveDefaults() {
 	 */
 	@Override
 	public Object proxy(Object target) {
-		AnnotationAwareOrderComparator.sort(this.advisors);
```

Comment:
I think we can likely do this without locking every time and possibly even without a lock.

Can you first try and implement `SmartInitializingBean#afterSingletons` move this sort call to that method? In this way, it will be called only once and also after the related method interceptors have been initialized.